src/Compilers/Core/Portable/Symbols/WellKnownMemberNames.cs

```diff
@@ -357,6 +357,6 @@ public static class WellKnownMemberNames
         /// <summary>
         /// The name of a type synthesized for a top-level statements entry point method.
         /// </summary>
-        public const string TopLevelStatementsEntryPointTypeName = "<Program>$";
+        public const string TopLevelStatementsEntryPointTypeName = "Program";
```

Comment:
FYI @davidwengier @tmat to see if this will break EnC:

https://github.com/dotnet/roslyn/blob/b46c7a752cfa94177dba13b836c7ccd5394d9043/src/Features/Core/Portable/EditAndContinue/AbstractEditAndContinueAnalyzer.cs#L5451-L5452

Reply:
Also FYI @CyrusNajmabadi, it's likely that generate field/property is going to break in a Program class:

https://github.com/dotnet/roslyn/blob/a07c7ff00fa58595250feb5c140193b9eba8d8d4/src/Features/Core/Portable/GenerateMember/GenerateVariable/AbstractGenerateVariableService.State.cs#L137-L140

Reply:
Thanks @Youssef1313. EnC itself should be fine, but I would expect test failures which will be easy enough to correct

Reply:
I don't think we can continue checking for top-level code container type just by name. We need to check if the `Program` type is in fact the synthesized implicit type (or if it is partial we need to somehow check if it has a synthesized partial declaration). It would be good to have a compiler API that tells us whether a type is the implicit top-level type.

There might also be some cases to test/consider around `partial class Program`.

Reply:
The distinctive mark of a simple program will be the `Program.<Main>$` method (remains unspeakable).

Reply:
So we'll need to check if the type contains `<Main>$`? Seems like there should be API that does that.

Reply:
The code was already checking for `<Main>$` so it's not really a change.
We can discuss possible public API, but I'm not clear what would the user scenarios be. Do we just care whether there is ` <Main>$`? Do we care whether it's a pure simple program, or whether it has some user-defined members from a partial declaration?

Reply:
I see. I guess it's ok then. For some reason I thought we are checking the type.